packages/text-editor/src/components/extension/validateContent.ts

```diff
@@ -0,0 +1,43 @@
+import { Editor, Extension } from '@tiptap/core'
+
+export interface ValidateContentOptions {
+  onEmpty?: (isEmpty: boolean) => void
+  onValidate?: (html: string) => void
+}
+
+export interface ValidateContentStorage {
+  isEmpty: boolean
+}
+
+const handleChange = (editor: Editor, options: ValidateContentOptions, storage: ValidateContentStorage): void => {
+  if (options.onEmpty !== undefined && options.onEmpty !== null) {
+    if (storage.isEmpty !== editor.isEmpty) {
+      storage.isEmpty = editor.isEmpty
+      options.onEmpty(storage.isEmpty)
+    }
+  }
+
+  if (options.onValidate !== undefined && options.onValidate !== null) {
+    options.onValidate(editor.getHTML()) // TODO: might need to call async ?
```

Comment:
Yes, maybe async ... 
When I tried to edit large documents `getHTML` was slow. Maybe we should pass editor so the calling code will decide how to perform validation (check html, prosemirror json, or text).

Reply:
I moved the extension for only `isEmpty` state, validation extension should probably be more complicated.